blockly/generators/arduino.js

```diff
@@ -98,23 +98,21 @@ Blockly.Arduino.init = function(workspace) {
   // Create a dictionary of setups to be printed before the code.
   Blockly.Arduino.setups_ = Object.create(null);
 
-  if (Blockly.Variables) {
-    if (!Blockly.Arduino.variableDB_) {
-      Blockly.Arduino.variableDB_ =
-          new Blockly.Names(Blockly.Arduino.RESERVED_WORDS_);
-    } else {
-      Blockly.Arduino.variableDB_.reset();
-    }
-
-    var defvars = [];
-    var variables = Blockly.Variables.allVariables(workspace);
-    for (var x = 0; x < variables.length; x++) {
-      defvars[x] = 'int ' +
-          Blockly.Arduino.variableDB_.getDistinctName(variables[x],
```

Comment:
getDistinctName was causing duplicate variable names.  Changed to getName (which is what core Blockly uses).